src/memory/AmdFlash.cc

```diff
@@ -26,7 +26,7 @@ AmdFlash::AmdFlash(const Rom& rom, Chip chip_,
 	: motherBoard(config.getMotherBoard())
 	, chip(chip_)
 	, sectorInfo(sectorInfo_)
-	, sz(sum(sectorInfo, &SectorInfo::size))
+	, sz(sum(chip.geometry.regions, [](Region r) { return r.count << r.size.exponent; }))
```

Comment:
Just an idea (I don't know yet if it's a good idea).
In `power_of_two` we could add something like this (not tested):
```
template<std::unsigned_integral T2> friend T2 operator*(T2 x, power_of_two<T> y) { return x << y; }
template<std::unsigned_integral T2> friend T2 operator*(power_of_two<T> x, T2 y) { return y << x; }
```
and then this usage can be simplified to just `return r.count * r.size`.

Reply:
I considered this, but thought I was maybe being too fancy and it was better to be explicit, so that you can see it’s a quick operation at the place where it’s used.

However now that you bring it up as well I will add these methods after all, because it _is_ pretty neat to do this and get an automatic optimisation even if you don’t think about it, and is more intuitive to read.